File: iOS_JavaScript/iOS-callJavaScript/OC&JavaScript/WKWebView-use/ViewController.m
//
//  ViewController.m
//  WKWebView-use
//
//  Created by 孟跃平 on 2017/6/13.
//  Copyright © 2017年 www.mengyueping.com. All rights reserved.
//

#import "ViewController.h"
#import <WebKit/WebKit.h>

@interface ViewController ()<WKNavigationDelegate, WKUIDelegate>

@end
@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    // app注册方法，供JS调用
    WKUserContentController *userContentController = [[WKUserContentController alloc] init];
    [userContentController addScriptMessageHandler:self name:@"messageHandler"];
    WKWebViewConfiguration *configuration = [[WKWebViewConfiguration alloc] init];
//    configuration.userContentController = userContentController;
    
    
    
    WKWebView *webView = [[WKWebView alloc] initWithFrame:self.view.bounds configuration:configuration];
    webView.UIDelegate = self;
    webView.navigationDelegate = self;
    [self.view addSubview:webView];
    NSURL *url = [NSURL URLWithString:@"https://www.baidu.com"];
    [webView loadRequest:[NSURLRequest requestWithURL:url]];
    
}


// WKNavigationDelegate
//首次载入调用顺序：didStartProvisionalNavigation -> didCommitNavigation -> didFinishNavigation
//重定向：didStartProvisionalNavigation -> didReceiveServerRedirectForProvisionalNavigation -> didCommitNavigation -> didFinishNavigation
- (void)webView:(WKWebView *)webView didStartProvisionalNavigation:(null_unspecified WKNavigation *)navigation {
    NSLog(@"--%s",__func__);
    //页面开始加载时调用
}
- (void)webView:(WKWebView *)webView didCommitNavigation:(null_unspecified WKNavigation *)navigation {
    NSLog(@"--%s",__func__);
    //当内容开始返回时调用
}
- (void)webView:(WKWebView *)webView didFinishNavigation:(null_unspecified WKNavigation *)navigation {
    NSLog(@"--%s",__func__);
    //页面加载完成之后调用
}


- (void)webView:(WKWebView *)webView didReceiveServerRedirectForProvisionalNavigation:(null_unspecified WKNavigation *)navigation {
    NSLog(@"--%s",__func__);
    //服务器重定向页面时调用,并且在 didStartProvisionalNavigation 之后，didCommitNavigation之前调用。
}
- (void)webView:(WKWebView *)webView didFailProvisionalNavigation:(null_unspecified WKNavigation *)navigation withError:(NSError *)error {
    NSLog(@"--%s",__func__);
    //页面加载失败时调用
}



@end

File: iOS_JavaScript/iOS-callJavaScript/Swift&JavaScript/WKWebView-use/ViewController.swift
//
//  ViewController.swift
//  WKWebView-use
//
//  Created by 孟跃平 on 2017/6/13.
//  Copyright © 2017年 www.mengyueping.com. All rights reserved.
//

import UIKit
import WebKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        let configuration = WKWebViewConfiguration()
        let webView = WKWebView(frame: view.bounds, configuration: configuration)
        webView.navigationDelegate = self
        webView.uiDelegate = self
        view.addSubview(webView)
        let url = URL(string: "https://www.baidu.com")!
        webView.load(URLRequest(url: url))
        
    }
}

extension ViewController: WKNavigationDelegate {
    func webView(_ webView: WKWebView, didStartProvisionalNavigation navigation: WKNavigation!) {
        print(#function)
        //页面开始加载时调用
    }
    func webView(_ webView: WKWebView, didCommit navigation: WKNavigation!) {
        print(#function)
        //当内容开始返回时调用
    }
    func webView(_ webView: WKWebView, didFinish navigation: WKNavigation!) {
        print(#function)
        //页面加载完成之后调用
    }
    func webView(_ webView: WKWebView, didReceiveServerRedirectForProvisionalNavigation navigation: WKNavigation!) {
        print(#function)
        //服务器重定向页面时调用,并且在 didStartProvisionalNavigation 之后，didCommitNavigation之前调用。
    }
    func webView(_ webView: WKWebView, didFailProvisionalNavigation navigation: WKNavigation!, withError error: Error) {
        print(#function)
        //页面加载失败时调用
    }
}
extension ViewController: WKUIDelegate {
    
}
